src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,582 @@
+            )
+            f_here = $(f_)(iv.here, qp, u, dof_range_here)
+            f_there = $(f_)(iv.there, qp, u, dof_range_there)
+            return $(is_avg ? :((f_here + f_there) / 2) : :(f_here - f_there))
+        end
+    end
+end
+
+function spatial_coordinate(iv::InterfaceValues, q_point::Int, x::NTuple{2, AbstractVector{<:Vec}}; here::Bool)
+    return here ? spatial_coordinate(iv.here, q_point, x[1]) : spatial_coordinate(iv.there, q_point, x[2])
+end
+
+
+# Transformation of quadrature points
+
+@doc raw"""
+    InterfaceTransformation
+
+Orientation information for 1D and 2D interfaces in 2D and 3D elements respectively.
+This information is used to construct the transformation matrix to
+transform the quadrature points from face_a to face_b achieving synced
+spatial coordinates. Face B's orientation relative to Face A's can
+possibly flipped (i.e. the vertices indices order is reversed)
+and the vertices can be rotated against each other.
+The reference orientation of face B is such that the first node
+has the lowest vertex index. Thus, this structure also stores the
+shift of the lowest vertex index which is used to reorient the face in
+case of flipping ["transform_interface_point!"](@ref).
+Take for example the faces
+```
+1           2
+| \         | \
+|  \        |  \
+| A \       | B \
+|    \      |    \
+2-----3     3-----1
+```
+which are rotated against each other by 240° after tranfroming to an
+equilateral triangle (shift index is 2) or the faces
+```
+2           2
+| \         | \
+|  \        |  \
+| A \       | B \
+|    \      |    \
+3-----1     3-----1
+```
+which are flipped against each other, note that face B has its reference node shifted by 2 indices
+so the face is tranformed into an equilateral triangle then rotated 120°, flipped about the x axis then
+rotated -120° and tranformed back to the reference triangle.Any combination of these can happen.
+"""
+struct InterfaceTransformation{RefShapeA, RefShapeB}
+    flipped::Bool
+    shift_index::Int
+    lowest_node_shift_index::Int
+    face_a::Int
+    face_b::Int
+end
+
+"""
+    InterfaceTransformation(cell_a::AbstractCell, cell_b::AbstractCell, face_a::Int, face_b::Int)
+
+Return the orientation info for the interface defined by face A and face B.
+"""
+function InterfaceTransformation(cell_a::AbstractCell, cell_b::AbstractCell, face_a::Int, face_b::Int)
+    getdim(cell_a) == 1 && return error("1D elements don't use transformations for interfaces.")
+
+    nodes_a = faces(cell_a)[face_a]
+    nodes_b = faces(cell_b)[face_b]
+
+    min_idx_a = argmin(nodes_a)
+    min_idx_b = argmin(nodes_b)
+
+    shift_index = min_idx_b - min_idx_a
+    flipped = getdim(cell_a) == 2 ? shift_index != 0 : nodes_a[min_idx_a != 1 ? min_idx_a - 1 : end] != nodes_b[min_idx_b != 1 ? min_idx_b - 1 : end]
+
+    return InterfaceTransformation{getrefshape(cell_a), getrefshape(cell_b)}(flipped, shift_index, 1 - min_idx_b, face_a, face_b)
+end
```

Comment:
Like in 
https://github.com/Ferrite-FEM/Ferrite.jl/blob/ce1886d9e36058052f5f79cc7bbf3586eb21869d/src/Grid/grid.jl#L870-L891 ?
This should be able to replace the other two orientation info structs in #829

Reply:
Indeed. But let us focus in this PR now on getting it merged, i.e. just resolving the open issues without touching more of the existing code base, to avoid feature creep.